### ISSUE TITLE
Potential breaking issue with continuous SMAX action space?

### ISSUE THREAD

**Issue by Chulabhaya** [FLAIROx/JaxMARL#160]
Hi there! I've been having trouble learning on the continuous action variant of SMAX (I'm exploring continuous action MARL problems), and doing a little deeper dive into the code for the environment led me to some sections which makes me think the current setup of the SMAX continuous action space has a few issues that might prevent anyone from learning an optimal policy.

In the SMAX class's `_decode_continuous_actions` function, there's this chunk of code (~lines 554- 567):

```
shoot_last_idx = self.continuous_action_dims.index("shoot_last_enemy")
        action_idx = self.continuous_action_dims.index("do_shoot")
        theta_idx = self.continuous_action_dims.index("coordinate_2")
        r_idx = self.continuous_action_dims.index("coordinate_1")
        shoot_last_enemy_logits = jnp.array(
            [
                jnp.log(actions[:, shoot_last_idx]),
                jnp.log(1 - actions[:, shoot_last_idx]),
            ]
        )
        logits = jnp.array(
            [jnp.log(actions[:, action_idx]), jnp.log(1 - actions[:, action_idx])]
        )
        move_or_shoot_key, shoot_last_enemy_key = jax.random.split(key)
        move_or_shoot = jax.random.categorical(move_or_shoot_key, logits, axis=0)
        shoot_last_enemy = jax.random.categorical(
            shoot_last_enemy_key, shoot_last_enemy_logits, axis=0
        )
```

As far as I understand, this says that even though the continuous action SMAX is defined with 4 continuous actions, the first two actions, `do_shoot` and `shoot_last_enemy`, are actually converted to discrete under the hood in the environment, where they are treating the continuous output as direct logits. This obviously has implications, e.g. when you're using MAPPO and you're calculating log probs/entropies for policy updates, because the environment action space is described as Continuous for all 4 dimensions, but in reality the first two are actually treated as Discrete actions. 

Beyond the issue that one might accidentally be using a Gaussian incorrectly for all 4 actions, and thus sampling the wrong log probs/entropies, there is also the issue that the environment itself is generating its own Categorical distributions and doing the sampling internally. This means I have no way in my policy updates to know what action was selected to get correct probabilities and entropies. I think what this means is that if I use, say MAPPO, then I'm always stuck to using incorrect log probs/entropies while doing my updates. Could anyone validate this for me or double check?

**Comment by Chulabhaya** [comment]
Another potential bug I think I've found: it appears that if you initialize a HeuristicEnemySMAX environment, its step_env function doesn't clip continuous actions to be between 0 and 1.

**Comment by Chulabhaya** [comment]
I took a stab at fixing some of my perceived issues here: [https://github.com/FLAIROx/JaxMARL/compare/main...Chulabhaya:JaxMARL:chulabhaya/smax_continuous_action_fixes](https://github.com/FLAIROx/JaxMARL/compare/main...Chulabhaya:JaxMARL:chulabhaya/smax_continuous_action_fixes)

The main changes are:
1. "Continuous" action space is properly changed to a hybrid space where the shoot/shoot-at-last-enemy are Discrete and the movement/targeting actions are Box.
2. Continuous actions are in the space [-1, 1], this is to make things more natural with the Gaussian in the neural network. Values are rescaled to be between 0 and 1 in the environment from the agent's raw [-1, 1] action
3. Radius value is scaled by a unit's attack range when calculating the targeting point. This is currently not being done which I think is a bug?

Would love to discuss this further, these particular changes lead to the orange plot compared to the baseline green plot in the 3m setup:

<img width="367" height="392" alt="Image" src="https://github.com/user-attachments/assets/16449da1-3668-4dbf-814b-72d970878bac" />
